persistence/scaladsl/src/main/scala/com/lightbend/lagom/scaladsl/persistence/PersistentEntityRegistry.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package com.lightbend.lagom.scaladsl.persistence
+
+import java.util.UUID
+
+import scala.concurrent.duration._
+import akka.stream.scaladsl
+import akka.NotUsed
+import akka.Done
+import scala.concurrent.Future
+
+/**
+ * At system startup all [[PersistentEntity]] classes must be registered here
+ * with [[PersistentEntityRegistry#register]].
+ *
+ * Later, [[PersistentEntityRef]] can be retrieved with [[PersistentEntityRegistry#refFor]].
+ * Commands are sent to a [[PersistentEntity]] using a `PersistentEntityRef`.
+ */
+trait PersistentEntityRegistry {
+
+  /**
+   * At system startup all [[PersistentEntity]] classes must be registered
+   * with this method.
+   */
+  def register[C, E, S](entityClass: Class[_ <: PersistentEntity[C, E, S]]): Unit
+
+  /**
+   * Retrieve a [[PersistentEntityRef]] for a given [[PersistentEntity]] class
+   * and identifier. Commands are sent to a `PersistentEntity` using a `PersistentEntityRef`.
+   */
+  def refFor[C](entityClass: Class[_ <: PersistentEntity[C, _, _]], entityId: String): PersistentEntityRef[C]
```

Comment:
Not for this PR, but I wonder how this can be done idiomatically... eg can we do something like (I think my use of `#` is wrong though):

```scala
def refFor[P <: PersistentEntity[_, _, _]: ClassTag](entityId: String): PersistentEntityRef[P#C]
```